sdk/openai/Azure.AI.OpenAI/src/Custom/ImageGenerations.cs

```diff
@@ -17,14 +17,25 @@ public partial class ImageGenerations
         /// <summary>
         ///     Gets a list of generated image items in the format specified for the request.
         /// </summary>
-        [CodeGenMemberSerializationHooks(DeserializationValueHook = nameof(DeserializeDataProperty))]
+        [CodeGenMemberSerializationHooks(SerializationValueHook = nameof(SerializeDataProperty), DeserializationValueHook = nameof(DeserializeDataProperty))]
         public IReadOnlyList<ImageLocation> Data { get; }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void SerializeDataProperty(Utf8JsonWriter writer)
+        {
+            // CUSTOM CODE NOTE: we always need to specify the serialization code explicitly when we change the type of a property
+            writer.WriteStartArray();
+            foreach (var item in Data)
+            {
+                writer.WriteObjectValue(item);
+            }
+            writer.WriteEndArray();
+        }
```

Comment:
we only need the deserialization hook because previously this model only have deserialization.
In public serialization feature, every model will have both serialization and deserialization therefore here we add the serialization hook as preparation for that to happen.
It currently does not have any effect to our generated code.